src/NLog/Internal/ExceptionHelper.cs

```diff
@@ -63,17 +155,35 @@ public static bool MustBeRethrown(this Exception exception)
                 return true;
             }
 
-            if (exception is NLogConfigurationException)
+            // only log after the 'serious' exceptions above
+
+            var shallRethrow = false;
+
+            if (!shallRethrow)
             {
-                return true;
+                shallRethrow = (exception is NLogConfigurationException) 
+                    || (exception.GetType().IsSubclassOf(typeof(NLogConfigurationException)))
+                    || LogManager.ThrowExceptions && !ignoreNonSevere;
             }
 
-            if (exception.GetType().IsSubclassOf(typeof(NLogConfigurationException)))
+            if(level == null)
             {
-                return true;
+                level = shallRethrow ? LogLevel.Error : LogLevel.Warn;
+
             }
 
-            return false;
+            if (level != LogLevel.Off)
+            {
+                if (string.IsNullOrEmpty(logMessage))
+                {
+                    InternalLogger.LogSecure(level, exception.Message);
```

Comment:
I think it's nice to have whole exception details in the log, so exception.toString()?